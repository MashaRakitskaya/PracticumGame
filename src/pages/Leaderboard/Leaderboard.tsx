import React, { useEffect } from 'react';
import LeaderboardItem from '../../components/LeaderboardItem/LeaderboardItem';
import { useFetchTeamLeaderboardMutation } from '../../services/leaderboard';
import { LeaderboardContainer, LeaderboardList } from './Leaderboard.styles.js';

const Leaderboard = () => {
  const [fetchLeaderboard, { data = [] }] = useFetchTeamLeaderboardMutation();

  useEffect(() => {
    fetchLeaderboard({ ratingFieldName: 'score', cursor: 0, limit: 100 });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <LeaderboardContainer role="leaderboard">
      <LeaderboardList>
        {data.map((item, index) => {
          return (
            <LeaderboardItem
              key={item.id}
              leaderboardItem={item}
              index={index}
            />
          );
        })}
      </LeaderboardList>
    </LeaderboardContainer>
  );
};

export default Leaderboard;
