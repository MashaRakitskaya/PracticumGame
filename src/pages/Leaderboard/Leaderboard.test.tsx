import React from 'react';
import { render } from '@testing-library/react';
import Leaderboard from './Leaderboard';

describe('Leaderboard component', () => {
  it('render Leaderboard', () => {
    const { asFragment } = render(<Leaderboard />);
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});
