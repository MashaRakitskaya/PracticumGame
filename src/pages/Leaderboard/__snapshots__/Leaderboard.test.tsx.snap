// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard component render Leaderboard 1`] = `
<DocumentFragment>
  <section
    class="css-8x7sxj"
  >
    <div
      class="css-ew27v6"
    >
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          1
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          104533200
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          2
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          10034420
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          3
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          1454234
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          4
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          1246434
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          5
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          1235634
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          6
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          1053500
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          7
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          1007740
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          8
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          167234
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          9
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          134234
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          10
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          124634
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          11
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          123456
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          12
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          123224
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          13
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          104600
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          14
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          104200
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          15
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          100460
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          16
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          100240
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          17
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1518020382113-a7e8fc38eac9?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=417&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          12324
        </p>
      </article>
      <article
        class="css-1jqozj7"
      >
        <span
          class="css-1qxwkg4"
        >
          18
        </span>
        <img
          alt="avatar"
          class="css-1r365q4"
          src="https://images.unsplash.com/photo-1517423568366-8b83523034fd?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=435&q=80"
        />
        <h3
          class="css-vldlm5"
        >
          Elena List
        </h3>
        <p
          class="css-cy7k6m"
        >
          10009
        </p>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
